xled/control.py

```diff
@@ -314,72 +586,289 @@ def set_led_movie_config(self, frame_delay, frames_number, leds_number):
         :param str password: password to use
         :raises ApplicationError: on application error
-        :rtype: None
+        :rtype: :class:`~xled.response.ApplicationResponse`
         """
-        assert self.hw_address
-        encpassword = encrypt_wifi_password(password, self.hw_address)
-        json_payload = {
-            "mode": 1,
-            "station": {"dhcp": 1, "ssid": ssid, "encpassword": encpassword},
-        }
+        json_payload = {"mode": 1}
+        if ssid and password:
+            assert self.hw_address
+            encpassword = xled.security.encrypt_wifi_password(password, self.hw_address)
+            json_payload["station"] = {
+                "dhcp": 1,
+                "ssid": ssid,
+                "encpassword": encpassword,
+            }
+        else:
+            assert not ssid and not password
         url = urljoin(self.base_url, "network/status")
         response = self.session.post(url, json=json_payload)
         app_response = ApplicationResponse(response)
         required_keys = [u"code"]
         assert all(key in app_response.keys() for key in required_keys)
+        return app_response
+
+    def set_playlist(self, entries):
+        """
+        Sets a new playlist
+
+        .. seealso:: :py:meth:`get_playlist()` :py:meth:`delete_playlist()`
+
+        :param list entries: list of playlist entries each with keys "unique_id" and "duration"
+        :raises ApplicationError: on application error
+        :rtype: :class:`~xled.response.ApplicationResponse`
+        """
+        assert isinstance(entries, list)
+        json_payload = {"entries": entries}
+        url = urljoin(self.base_url, "playlist")
+        response = self.session.post(url, json=json_payload)
+        app_response = ApplicationResponse(response)
+        required_keys = [u"code"]
+        assert all(key in app_response.keys() for key in required_keys)
+        return app_response
+
+    def set_playlist_current(self, movie_id):
+        """
+        Sets which movie in the playlist to play
+
+        .. seealso:: :py:meth:`get_playlist_current()`
+
+        :raises ApplicationError: on application error
+        :rtype: :class:`~xled.response.ApplicationResponse`
+        """
+        json_payload = {"id": movie_id}
+        url = urljoin(self.base_url, "playlist/current")
+        response = self.session.post(url, json=json_payload)
+        app_response = ApplicationResponse(response)
+        required_keys = [u"code"]
+        assert all(key in app_response.keys() for key in required_keys)
+        return app_response
+
+    def set_rt_frame_rest(self, frame):
+        """
+        Uploads a frame in rt-mode, using the ordinary restful protocol
+
+        :param frame: file-like object that points to frame file.
+        :raises ApplicationError: on application error
+        :rtype: :class:`~xled.response.ApplicationResponse`
+        """
+        url = urljoin(self.base_url, "led/rt/frame")
```

Comment:
I'm sorry that I've missed much of the discussion here. This looks interesting! Is it a change with new firmware versions to support this? How did you find out this call existed?

Reply:
@magicus see https://xled-docs.readthedocs.io/en/latest/rest_api.html#send-realtime-frame . I found that https://github.com/scrool/xled-docs/commit/4dc705574ac166f10b1bf7e89a1c50a86f163371 , I believe by watching data on a wire.

Reply:
I see, thanks. I have not been following the updates to the docs.